tensorflow_addons/seq2seq/basic_decoder_test.py

```diff
@@ -122,6 +122,132 @@ def testStepWithTrainingHelperOutputLayer(self, use_output_layer):
+        input_depth = 7
+        cell_depth = 10
+        output_layer_depth = 3
+
+        with self.cached_session(use_gpu=True):
+            inputs = np.random.randn(batch_size, max_time,
+                                     input_depth).astype(np.float32)
+            input_t = tf.constant(inputs)
+            cell = tf.keras.layers.LSTMCell(cell_depth)
+            sampler = sampler_py.TrainingSampler(time_major=False)
+            output_layer = tf.keras.layers.Dense(
+                output_layer_depth, use_bias=False)
+            expected_output_depth = output_layer_depth
+            initial_state = cell.get_initial_state(
+                batch_size=batch_size, dtype=tf.float32)
+            my_decoder = basic_decoder.BasicDecoder(
+                cell=cell, sampler=sampler, output_layer=output_layer)
+
+            if use_mask:
+                (first_finished, first_inputs,
+                 first_state) = my_decoder.initialize(
+                     input_t, initial_state=initial_state, mask=mask)
+            else:
+                (first_finished, first_inputs,
+                 first_state) = my_decoder.initialize(
+                     input_t,
+                     initial_state=initial_state,
+                     sequence_length=sequence_length)
+
+            output_size = my_decoder.output_size
+            output_dtype = my_decoder.output_dtype
+            self.assertEqual(
+                basic_decoder.BasicDecoderOutput(expected_output_depth,
+                                                 tf.TensorShape([])),
+                output_size)
+            self.assertEqual(
+                basic_decoder.BasicDecoderOutput(tf.float32, tf.int32),
+                output_dtype)
+
+            (step_outputs, step_state, step_next_inputs,
+             step_finished) = my_decoder.step(
+                 tf.constant(0), first_inputs, first_state)
+            batch_size_t = my_decoder.batch_size
+
+            self.assertLen(first_state, 2)
+            self.assertLen(step_state, 2)
+            self.assertIsInstance(step_outputs,
+                                  basic_decoder.BasicDecoderOutput)
+            self.assertEqual((batch_size, expected_output_depth),
+                             step_outputs[0].get_shape())
+            self.assertEqual((batch_size,), step_outputs[1].get_shape())
+            self.assertEqual((batch_size, cell_depth),
+                             first_state[0].get_shape())
+            self.assertEqual((batch_size, cell_depth),
+                             first_state[1].get_shape())
+            self.assertEqual((batch_size, cell_depth),
+                             step_state[0].get_shape())
+            self.assertEqual((batch_size, cell_depth),
+                             step_state[1].get_shape())
+
+            self.assertEqual(len(output_layer.variables), 1)
+
+            self.evaluate(tf.compat.v1.global_variables_initializer())
```

Comment:
Assume we use the tf 2.0 as dependency, I guess this is no longer necessary?

Reply:
Yes, you're right, I fixed it in the new commits.  However, I used the previous tests as a base for this one, so I guess we need to update them too (in a separate PR though)